### ISSUE TITLE
has no attribute 'ActionBase' error upon ansible playbook run (Mitogen issue potentially)

### ISSUE THREAD

**Issue by TheCorp** [ansible-collections/community.mongodb#434]
##### SUMMARY
I am able to run my playbook with just `community.mongodb.mongodb_repository` but once I include `community.mongodb.mongodb_install` then it throws the has no attribute 'ActionBase' error. I have a feeling this has to do with Mitogen and I looked into the existing issues but couldn't find anything like this error. I'm hoping this is just me doing something dumb...

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME
mongodb_install

##### ANSIBLE VERSION

```
ansible 2.9.27
  config file = /mnt/sf_workspace/central/ansible/ansible.cfg
  configured module search path = ['/mnt/sf_workspace/ansible/modules', '/mnt/sf_workspace/central/ansible/modules']
  ansible python module location = /usr/local/lib/python3.8/dist-packages/ansible
  executable location = /usr/local/bin/ansible
  python version = 3.8.10 (default, Nov 26 2021, 20:14:08) [GCC 9.3.0]
```

##### COLLECTION VERSION

```
This didn't seem to work... 
usage: ansible-galaxy collection [-h] COLLECTION_ACTION ...
ansible-galaxy collection: error: argument COLLECTION_ACTION: invalid choice: 'list' (choose from 'init', 'build', 'publish', 'install')

```

##### CONFIGURATION

```ANSIBLE_PIPELINING(/mnt/sf_workspace/central/ansible/ansible.cfg) = True
ANSIBLE_SSH_ARGS(/mnt/sf_workspace/central/ansible/ansible.cfg) = -o UserKnownHostsFile=/dev/null -o ControlMaster=auto -o ControlPersist=3600s -o ControlPath=~/.ansible/ssh-%h -o StrictHostKeyChecking=no
ANSIBLE_SSH_RETRIES(/mnt/sf_workspace/central/ansible/ansible.cfg) = 2
COLLECTIONS_PATHS(/mnt/sf_workspace/central/ansible/ansible.cfg) = ['/mnt/sf_workspace/ansible/galaxy_collections', '/mnt/sf_workspace/central/ansible/galaxy_collections']
DEFAULT_CALLBACK_PLUGIN_PATH(/mnt/sf_workspace/central/ansible/ansible.cfg) = ['/mnt/sf_workspace/ansible/callbacks', '/mnt/sf_workspace/central/ansible/callbacks']
DEFAULT_CALLBACK_WHITELIST(/mnt/sf_workspace/central/ansible/ansible.cfg) = ['log_plays2']
DEFAULT_FORCE_HANDLERS(/mnt/sf_workspace/central/ansible/ansible.cfg) = True
DEFAULT_FORKS(/mnt/sf_workspace/central/ansible/ansible.cfg) = 100
DEFAULT_HOST_LIST(/mnt/sf_workspace/central/ansible/ansible.cfg) = ['/mnt/sf_workspace/central/ansible/hosts/local']
DEFAULT_MODULE_PATH(/mnt/sf_workspace/central/ansible/ansible.cfg) = ['/mnt/sf_workspace/ansible/modules', '/mnt/sf_workspace/central/ansible/modules']
DEFAULT_REMOTE_USER(/mnt/sf_workspace/central/ansible/ansible.cfg) = ubuntu
DEFAULT_ROLES_PATH(/mnt/sf_workspace/central/ansible/ansible.cfg) = ['/mnt/sf_workspace/ansible/galaxy_roles', '/mnt/sf_workspace/ansible/roles', '/mnt/sf_workspace/central/ansible/galaxy_roles', '/mnt/s>
DEFAULT_STRATEGY(/mnt/sf_workspace/central/ansible/ansible.cfg) = mitogen_linear
DEFAULT_STRATEGY_PLUGIN_PATH(/mnt/sf_workspace/central/ansible/ansible.cfg) = ['/mnt/sf_workspace/ansible/plugins/mitogen-0.2.9/ansible_mitogen/plugins/strategy', '/mnt/sf_workspace/central/ansible/plugi>
HOST_KEY_CHECKING(/mnt/sf_workspace/central/ansible/ansible.cfg) = False
RETRY_FILES_ENABLED(/mnt/sf_workspace/central/ansible/ansible.cfg) = False
TRANSFORM_INVALID_GROUP_CHARS(/mnt/sf_workspace/central/ansible/ansible.cfg) = ignore

```

##### OS / ENVIRONMENT
Ubuntu 20.20 LTS

##### STEPS TO REPRODUCE

```sudo ansible-playbook playbooks/central/main.yml -vvvv

```

##### EXPECTED RESULTS
Playbook should run without issue

##### ACTUAL RESULTS

```ansible-playbook 2.9.27
  config file = /mnt/sf_workspace/central/ansible/ansible.cfg
  configured module search path = ['/mnt/sf_workspace/ansible/modules', '/mnt/sf_workspace/central/ansible/modules']
  ansible python module location = /usr/local/lib/python3.8/dist-packages/ansible
  executable location = /usr/local/bin/ansible-playbook
  python version = 3.8.10 (default, Nov 26 2021, 20:14:08) [GCC 9.3.0]
Using /mnt/sf_workspace/central/ansible/ansible.cfg as config file
setting up inventory plugins
host_list declined parsing /mnt/sf_workspace/central/ansible/hosts/local as it did not pass its verify_file() method
script declined parsing /mnt/sf_workspace/central/ansible/hosts/local as it did not pass its verify_file() method
auto declined parsing /mnt/sf_workspace/central/ansible/hosts/local as it did not pass its verify_file() method
Set default localhost to localhost
Parsed /mnt/sf_workspace/central/ansible/hosts/local inventory source with ini plugin
Read vars_file '../../../../ansible/group_vars/base'
Read vars_file '../../group_vars/central'
Read vars_file '../../../../ansible/group_vars/base'
Read vars_file '../../group_vars/central'
Read vars_file '../../../../ansible/group_vars/base'
Read vars_file '../../group_vars/central'
Read vars_file '../../../../ansible/group_vars/base'
Read vars_file '../../group_vars/central'
Read vars_file '../../../../ansible/group_vars/base'
Read vars_file '../../group_vars/central'
Read vars_file '../../../../ansible/group_vars/base'
Read vars_file '../../group_vars/central'
Read vars_file '../../../../ansible/group_vars/base'
Read vars_file '../../group_vars/central'
Read vars_file '../../../../ansible/group_vars/base'
Read vars_file '../../group_vars/central'
Read vars_file '../../../../ansible/group_vars/base'
Read vars_file '../../group_vars/central'
Read vars_file '../../../../ansible/group_vars/base'
Read vars_file '../../group_vars/central'
Read vars_file '../../../../ansible/group_vars/base'
Read vars_file '../../group_vars/central'
Read vars_file '../../../../ansible/group_vars/base'
Read vars_file '../../group_vars/central'
Read vars_file '../../../../ansible/group_vars/base'
Read vars_file '../../group_vars/central'
Read vars_file '../../../../ansible/group_vars/base'
Read vars_file '../../group_vars/central'
Read vars_file '../../../../ansible/group_vars/base'
Read vars_file '../../group_vars/central'
Read vars_file '../../../../ansible/group_vars/base'
Read vars_file '../../group_vars/central'
Read vars_file '../../../../ansible/group_vars/base'
Read vars_file '../../group_vars/central'
statically imported: /mnt/sf_workspace/ansible/roles/base/tasks/ssh.yml
Read vars_file '../../../../ansible/group_vars/base'
Read vars_file '../../group_vars/central'
statically imported: /mnt/sf_workspace/ansible/roles/base/tasks/ec2.yml
Read vars_file '../../../../ansible/group_vars/base'
Read vars_file '../../group_vars/central'
statically imported: /mnt/sf_workspace/ansible/roles/base/tasks/storage.yml
Read vars_file '../../../../ansible/group_vars/base'
Read vars_file '../../group_vars/central'
Read vars_file '../../../../ansible/group_vars/base'
Read vars_file '../../group_vars/central'
statically imported: /mnt/sf_workspace/ansible/roles/base/tasks/swap.yml
Read vars_file '../../../../ansible/group_vars/base'
Read vars_file '../../group_vars/central'
statically imported: /mnt/sf_workspace/ansible/roles/base/tasks/directories.yml
Read vars_file '../../../../ansible/group_vars/base'
Read vars_file '../../group_vars/central'
statically imported: /mnt/sf_workspace/ansible/roles/base/tasks/logs.yml
Read vars_file '../../../../ansible/group_vars/base'
Read vars_file '../../group_vars/central'
statically imported: /mnt/sf_workspace/ansible/roles/base/tasks/symlinks.yml
Read vars_file '../../../../ansible/group_vars/base'
Read vars_file '../../group_vars/central'
statically imported: /mnt/sf_workspace/ansible/roles/base/tasks/ssl.yml
Read vars_file '../../../../ansible/group_vars/base'
Read vars_file '../../group_vars/central'
statically imported: /mnt/sf_workspace/ansible/roles/base/tasks/tools.yml
Read vars_file '../../../../ansible/group_vars/base'
Read vars_file '../../group_vars/central'
statically imported: /mnt/sf_workspace/ansible/roles/base/tasks/profile.yml
Read vars_file '../../../../ansible/group_vars/base'
Read vars_file '../../group_vars/central'
statically imported: /mnt/sf_workspace/ansible/roles/base/tasks/chrony.yml
Read vars_file '../../../../ansible/group_vars/base'
Read vars_file '../../group_vars/central'
statically imported: /mnt/sf_workspace/ansible/roles/base/tasks/purge.yml
Read vars_file '../../../../ansible/group_vars/base'
Read vars_file '../../group_vars/central'
statically imported: /mnt/sf_workspace/ansible/roles/base/tasks/orphans.yml
Read vars_file '../../../../ansible/group_vars/base'
Read vars_file '../../group_vars/central'
statically imported: /mnt/sf_workspace/ansible/roles/base/tasks/cron.yml
Read vars_file '../../../../ansible/group_vars/base'
Read vars_file '../../group_vars/central'
statically imported: /mnt/sf_workspace/ansible/roles/base/tasks/datadog-agent.yml
Read vars_file '../../../../ansible/group_vars/base'
Read vars_file '../../group_vars/central'
statically imported: /mnt/sf_workspace/ansible/roles/base/tasks/datadog-monitors.yml
Read vars_file '../../../../ansible/group_vars/base'
Read vars_file '../../group_vars/central'
statically imported: /mnt/sf_workspace/ansible/roles/base/tasks/dump-vars.yml
Read vars_file '../../../../ansible/group_vars/base'
Read vars_file '../../group_vars/central'
Read vars_file '../../../../ansible/group_vars/base'
Read vars_file '../../group_vars/central'
Read vars_file '../../../../ansible/group_vars/base'
Read vars_file '../../group_vars/central'
Loading callback plugin default of type stdout, v2.0 from /usr/local/lib/python3.8/dist-packages/ansible/plugins/callback/default.py
Skipping callback 'actionable', as we already have a stdout callback.
Skipping callback 'counter_enabled', as we already have a stdout callback.
Skipping callback 'debug', as we already have a stdout callback.
Skipping callback 'dense', as we already have a stdout callback.
Skipping callback 'dense', as we already have a stdout callback.
Skipping callback 'full_skip', as we already have a stdout callback.
Skipping callback 'json', as we already have a stdout callback.
Loading callback plugin log_plays2 of type notification, v2.1 from /mnt/sf_workspace/ansible/callbacks/log_plays2.py
Skipping callback 'minimal', as we already have a stdout callback.
Skipping callback 'null', as we already have a stdout callback.
Skipping callback 'oneline', as we already have a stdout callback.
Skipping callback 'selective', as we already have a stdout callback.
Skipping callback 'skippy', as we already have a stdout callback.
Skipping callback 'stderr', as we already have a stdout callback.
Skipping callback 'unixy', as we already have a stdout callback.
Skipping callback 'yaml', as we already have a stdout callback.

PLAYBOOK: main.yml *****************************************************************************************************************************************************************************************
Positional arguments: playbooks/central/main.yml
verbosity: 4
remote_user: ubuntu
connection: smart
timeout: 10
force_handlers: True
become_method: sudo
tags: ('all',)
inventory: ('/mnt/sf_workspace/central/ansible/hosts/local',)
forks: 100
1 plays in playbooks/central/main.yml
Read vars_file '../../../../ansible/group_vars/base'
Read vars_file '../../group_vars/central'
Read vars_file '../../../../ansible/group_vars/base'
Read vars_file '../../group_vars/central'
Read vars_file '../../../../ansible/group_vars/base'
Read vars_file '../../group_vars/central'

PLAY [central] *********************************************************************************************************************************************************************************************
ERROR! Unexpected Exception, this is probably a bug: module 'ansible_collections.ansible.builtin.plugins.action' has no attribute 'ActionBase'
the full traceback was:

Traceback (most recent call last):
  File "/usr/local/bin/ansible-playbook", line 123, in <module>
    exit_code = cli.run()
  File "/usr/local/lib/python3.8/dist-packages/ansible/cli/playbook.py", line 128, in run
    results = pbex.run()
  File "/usr/local/lib/python3.8/dist-packages/ansible/executor/playbook_executor.py", line 169, in run
    result = self._tqm.run(play=play)
  File "/usr/local/lib/python3.8/dist-packages/ansible/executor/task_queue_manager.py", line 261, in run
    strategy = strategy_loader.get(new_play.strategy, self)
  File "/usr/local/lib/python3.8/dist-packages/ansible/plugins/loader.py", line 552, in get
    self._module_cache[path] = self._load_module_source(name, path)
  File "/usr/local/lib/python3.8/dist-packages/ansible/plugins/loader.py", line 525, in _load_module_source
    spec.loader.exec_module(module)
  File "<frozen importlib._bootstrap_external>", line 848, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/mnt/sf_workspace/ansible/plugins/mitogen-0.2.9/ansible_mitogen/plugins/strategy/mitogen_linear.py", line 56, in <module>
    import ansible_mitogen.strategy
  File "/mnt/sf_workspace/ansible/plugins/mitogen-0.2.9/ansible_mitogen/strategy.py", line 43, in <module>
    import ansible_mitogen.mixins
  File "/mnt/sf_workspace/ansible/plugins/mitogen-0.2.9/ansible_mitogen/mixins.py", line 67, in <module>
    class ActionModuleMixin(ansible.plugins.action.ActionBase):
AttributeError: module 'ansible_collections.ansible.builtin.plugins.action' has no attribute 'ActionBase'

```

**Comment by TheCorp** [comment]
I think its possibly because of this - https://github.com/mitogen-hq/mitogen/issues/652 and because the version of Mitogen we're on doesn't support collections (i'm trying to add community.mongodb) but it wasnt clear from the errors if these were definitely related issues. Apologies if I'm just being dense about this!

**Comment by rhysmeister** [comment]
Hello @TheCorp,

No problem, yes I would agree it looks like a mitogen issue. I tried mitogen a while back because it seemed a pretty cool way to speed stuff up. I didn't have a good experience and ended up uninstalling it. They mention on this issue.. https://github.com/cytopia/docker-ansible/issues/96 that a fix way made for 2.10 so upgrading might be worth a shot.

Cheers,

Rhys

**Comment by TheCorp** [comment]
Thanks Rhys, we can close this out. We're not ready to upgrade to an unreleased version of mitogen so we'll test this out and collections later once thats fully released. Appreciate the assistance!
